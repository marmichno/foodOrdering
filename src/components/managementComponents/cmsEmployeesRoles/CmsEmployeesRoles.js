import {CmsNavbar} from '../cmsNavbar/CmsNavbar';
import {useState, useEffect} from 'react';
import {employeesRolesGetRequest} from '../../../requests/employeesRolesGetRequest';
import {employeesRolesPostRequest} from '../../../requests/employeesRolesPostRequest';
import {employeesRolesDeleteRequest} from '../../../requests/employeesRolesDeleteRequest';

export const CmsEmployeesRoles = () =>{

    const [roles, setRoles] = useState([]);
    const [roleName, setRoleName] = useState("");

    useEffect(() =>{
        getRoles();
    },[]);

    const getRoles = async () =>{
        const response = await employeesRolesGetRequest();
        setRoles(response);
    }

    const addRole = async () => {
        const response = await employeesRolesPostRequest({
            "name":roleName
        });
        getRoles();
    }

    const deleteRole = async (e) => {
        const response = await employeesRolesDeleteRequest(e.target.dataset.id);
        getRoles();
    }

    return(
        <div className="employeesRolesMainContainer">
            <CmsNavbar/>
            <div className="employeesRolesContent">
                <div className="roleRow"></div>
                <div className="roleRow">
                    <div><h1>id</h1></div>
                    <div><h1>Role name</h1></div>
                    <div><h1>Employees in role</h1></div>
                    <div><h1>Edit</h1></div>
                </div>

                <div className="addRoleContainer">
                    <input onChange={(e) => setRoleName(e.target.value)} type="text" placeholder="group name" autocomplete="off" name="category" required></input>
                    <button onClick={addRole}>Add employee role</button>
                </div>

                {
                roles !== undefined ?
                    roles.map(value => {
                        return(
                            <div className="roleRow">
                                <div><p>{value.id}</p></div>
                                <div><p>{value.name}</p></div>
                                <div><p>123</p></div>
                                <div><button>modify</button><button data-id={value.id} onClick={deleteRole}>delete</button></div>
                            </div>
                        )
                    })
                : null
                }
            </div>
        </div>
    )
}