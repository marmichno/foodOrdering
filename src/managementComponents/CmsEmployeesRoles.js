import {CmsNavbar} from './CmsNavbar';
import {useState, useEffect} from 'react';
import {employeesRolesGetRequest} from '../requests/employeesRolesGetRequest';
import {employeesRolesPostRequest} from '../requests/employeesRolesPostRequest';

export const CmsEmployeesRoles = () =>{

    const [roles, setRoles] = useState("");
    const [roleName, setRoleName] = useState("");

    useEffect(() =>{
        getRoles();
    },[]);

    const getRoles = async () =>{
        const response = await employeesRolesGetRequest();
        setRoles(response);
    }

    const addRole = async () => {
        const response = await employeesRolesPostRequest({
            "name":roleName
        });
        getRoles();
    }

    return(
        <div className="employeesRolesMainContainer">
            <CmsNavbar/>
            <div className="employeesRolesContent">
                <div className="roleRow">
                    <div><h1>id</h1></div>
                    <div><h1>Role name</h1></div>
                    <div><h1>Employees in role</h1></div>
                    <div><h1>Edit</h1></div>
                </div>

                <div className="addRoleContainer">
                    <input onChange={(e) => setRoleName(e.target.value)} type="text" placeholder="group name" autocomplete="off" name="category" required></input>
                    <button onClick={addRole}>Add employee role</button>
                </div>

                {
                roles !== undefined ?
                    roles.map(value => {
                        return(
                            <div className="groupRole">
                                <div><p>{value.id}</p></div>
                                <div><p>{value.name}</p></div>
                                <div><p>123</p></div>
                                <div><button>modify</button><button>delete</button></div>
                            </div>
                        )
                    })
                : null
                }
            </div>
        </div>
    )
}