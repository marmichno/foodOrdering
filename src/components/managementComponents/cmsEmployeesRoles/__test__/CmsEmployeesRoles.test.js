jest.unmock('axios');
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {CmsEmployeesRoles} from '../CmsEmployeesRoles';
import "@testing-library/jest-dom/extend-expect";
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux'; 
import validEmployeesRolesJson from '../__mocks__/validEmployeesRoles.json';
import TestRenderer from "react-test-renderer";

const store = createStore(() => [], {}, applyMiddleware());

const mock = new MockAdapter(axios);

test("checks if employees roles are visible after api call", async () => {

    mock.onGet("http://localhost:8080/api/role").reply(200, validEmployeesRolesJson);

    const {getAllByText} = render(<Provider store={store}><CmsEmployeesRoles/></Provider>);

    await waitFor(() => {
        expect(getAllByText('sushi chef'));
        expect(getAllByText('123'));
    })
});

test("matches snapshot", () => {
    const tree = TestRenderer.create(<Provider store={store}><CmsEmployeesRoles/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});
