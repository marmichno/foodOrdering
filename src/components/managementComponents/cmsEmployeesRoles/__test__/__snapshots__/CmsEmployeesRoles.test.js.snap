// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="employeesRolesMainContainer"
>
  <div
    className="hamburgerMenu"
  >
    <svg
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="employeesRolesContent"
  >
    <div
      className="roleRow"
    />
    <div
      className="roleRow"
    >
      <div>
        <h1>
          id
        </h1>
      </div>
      <div>
        <h1>
          Role name
        </h1>
      </div>
      <div>
        <h1>
          Employees in role
        </h1>
      </div>
      <div>
        <h1>
          Edit
        </h1>
      </div>
    </div>
    <div
      className="addRoleContainer"
    >
      <input
        autoComplete="off"
        name="category"
        onChange={[Function]}
        placeholder="group name"
        required={true}
        type="text"
      />
      <button
        onClick={[Function]}
      >
        Add employee role
      </button>
    </div>
  </div>
</div>
`;
